File: src/app/browser/header/tab.cpp
#include "tab.hpp"

using namespace app::browser::header;

Tab::Tab(
    const Glib::RefPtr<Gio::SimpleAction> & ACTION__TAB_APPEND
) {
    // Init actions
    action__tab_append = ACTION__TAB_APPEND;

    // Init widget
    set_icon_name(
        "tab-new-symbolic"
    );

    set_tooltip_text(
        _("New tab")
    );

    // Init events
    signal_clicked().connect(
        [this]
        {
            action__tab_append->activate();
        }
    );
}

File: src/app/browser/window/header/tray/tab.rs
use gtk::{gio::SimpleAction, prelude::ActionExt, prelude::ButtonExt, Button};
use std::sync::Arc;

pub struct Tab {
    pub gobject: Button,
}

impl Tab {
    // Construct
    pub fn new(action_tab_append: Arc<SimpleAction>) -> Self {
        // Init widget
        let gobject = Button::builder()
            .icon_name("tab-new-symbolic")
            .tooltip_text("New tab")
            .build();

        // Init events
        gobject.connect_clicked(move |_| {
            action_tab_append.activate(None);
        });

        // Return activated struct
        Self { gobject }
    }

    // Getters
    pub fn gobject(&self) -> &Button {
        &self.gobject
    }
}
